app/src/main/java/com/itsaky/androidide/activities/MainActivity.kt

```diff
@@ -109,10 +112,24 @@ class MainActivity : IDEActivity() {
       }
     }
 
-    viewModel.setScreen(SCREEN_MAIN)
+    Log.d("MainActivity", "Current screen: ${viewModel.currentScreen.value}")
+    Log.d("MainActivity", "Previous screen: ${viewModel.previousScreen}")
+
+    // Data in a ViewModel is kept between activity rebuilds on
+    // configuration changes (i.e. screen rotation)
+    // * previous == -1 and current == -1 -> this is an initial instantiation of the activity
+    if (viewModel.currentScreen.value == -1 && viewModel.previousScreen == -1) {
+      viewModel.setScreen(SCREEN_MAIN)
+    } else {
+      onScreenChanged(viewModel.currentScreen.value)
+    }
 
-    onBackPressedDispatcher.addCallback(/* owner = */
-      this, /* onBackPressedCallback = */ onBackPressedCallback)
+    onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
+  }
+
+  override fun onInsetsUpdated(insets: Rect) {
+    super.onInsetsUpdated(insets)
+    _binding!!.fragmentContainersParent.setPadding(insets.left, 0, insets.right, insets.bottom)
```

Comment:
You can use `binding` instead of `_binding!!`.

https://github.com/AndroidIDEOfficial/AndroidIDE/blob/a9a24242af34ed0cab06d7479c42b05ee410b821/app/src/main/java/com/itsaky/androidide/activities/MainActivity.kt#L83-L84